codec-ohttp/src/main/java/io/netty/incubator/codec/ohttp/OHttpCrypto.java

```diff
@@ -67,33 +47,30 @@ private static ByteBuffer readableTemporaryBuffer(ByteBuf buf, int length) {
      * Encrypt a message of a given length and write the encrypted data to a buffer.
      *
      * @param message           the message
-     * @param messageLength     the length to encrypt
+     * @param messageLength     the length of the message
      * @param isFinal           {@code true} if this is the final message.
      * @param out               {@link ByteBuf} into which the encrypted data is written.
      * @throws CryptoException  thrown when an error happens.
      */
     public final void encrypt(ByteBuf message, int messageLength, boolean isFinal, ByteBuf out) throws CryptoException {
-        final ByteBuffer encrypted = encryptCrypto().seal(
-                aad(isFinal && configuration().useFinalAad()),
-                readableTemporaryBuffer(message, messageLength));
+        encryptCrypto().seal(aad(isFinal && configuration().useFinalAad()),
+                message.slice(message.readerIndex(), messageLength), out);
```

Comment:
Should we use `message.readSlice()` and remove `message.skipBytes()`?

Reply:
I was not sure if we should better only increase the reader index if we dont throw an exception. Let's keep it like this to be safe